shell/common/engine.h

```diff
@@ -790,10 +797,10 @@ class Engine final : public RuntimeDelegate,
   ///             The Dart compiler may generate separate shared libraries
   ///             files called 'loading units' when libraries are imported
   ///             as deferred. Each of these shared libraries are identified
-  ///             by a unique loading unit id. Callers should dlopen the
-  ///             shared library file and use dlsym to resolve the dart
-  ///             symbols. These symbols can then be passed to this method to
-  ///             be dynamically loaded into the VM.
+  ///             by a unique loading unit id. Callers should open and resolve
+  ///             a SymbolMapping from the shared library. The symbols can
```

Comment:
If you're being explicit about taking a `SymbolMapping`, should the signature be specified to take `SymbolMapping` too?

Reply:
`SymbolMapping` is used when we own the the native library handle too. In some cases (usually where the platform or embedder owns this handle), we can use an `UnownedMapping`. In any case, both are subclasses of `fml::Mapping`.  I don't think this layer should care how the mapping is resolved. Just that there is a mapping whose lifecycle has been sorted out (and readers can follow).